data_transfer/subset_matrix_table.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+"""
+Takes a path to a MatrixTable, and optionally sample IDs
+Reads in the MT from provided input path
+If sample IDs are specified, subset containing only those
+Write the [reduced] MT into Test
+Optionally with the --vcf flag, write as a VCF
+Optionally with both --chr and --pos specified, subset to a specific locus
+"""
+
+from argparse import ArgumentParser
+import logging
+import sys
+
+import hail as hl
+
+from cpg_utils.hail_batch import output_path, init_batch
+from cpg_utils.config import get_config
+
+
+def subset_to_samples(
+    matrix: hl.MatrixTable, samples: list[str], remove_hom_ref: bool
+) -> hl.MatrixTable:
+    """
+    reduce the MatrixTable to a subset
+    Parameters
+    ----------
+    matrix :
+    samples :
+    remove_hom_ref :
+
+    Returns
+    -------
+
+    """
+
+    mt_samples = matrix.s.collect()
+
+    # check the subset exists
+    missing_samples = [sam for sam in samples if sam not in mt_samples]
+    if missing_samples:
+        raise Exception(f"Sample(s) missing from subset: {','.join(missing_samples)}")
+
+    filt_mt = matrix.filter_cols(hl.literal(set(samples)).contains(matrix.s))
+
+    # optional - filter to variants with at least one alt call in these samples
+    if remove_hom_ref:
+        filt_mt = filt_mt.filter_entries(filt_mt.GT.is_non_ref())
+
+    return filt_mt
+
+
+def subset_to_locus(matrix: hl.MatrixTable, chrom: str, pos: int) -> hl.MatrixTable:
+    """
+
+    Parameters
+    ----------
+    matrix :
+    chrom :
+    pos :
+
+    Returns
+    -------
+
+    """
+
+    locus = hl.Locus(contig=chrom, position=pos, reference_genome="GRCh38")
```

Comment:
Here and below: we typically use single quotes for strings, not double quotes.

Reply:
Aye, I have a file watcher for Black, but it doesn't have the standard pre-commit settings... I'll resolve that

Reply:
updated repo to include python-standard pre-commit hooks